core/dbt/include/global_project/macros/materializations/tests/helpers.sql

```diff
@@ -3,12 +3,27 @@
 {%- endmacro %}
 
 {% macro default__get_test_sql(main_sql, fail_calc, warn_if, error_if, limit) -%}
+    {% set model_name = model.file_key_name.split('.')[1] %}
```

Comment:
Two thoughts:
- For tests defined on models, this should work. For tests defined on sources, `file_key_name` is `sources.source_name` (and missing the name of the specific source table), so I don't think it would work as currently constituted. We could update `file_key_name` to include `sources.source_name.source_table`.
- If users have defined a `where` config, and we're calculating `%`, I think we would want to filter both the numerator and denominator by that `where` config. You could reproduce that logic here (`config.get("where")`), but all it all it seems preferable if there were some where to access the rendered version of `{{ model }}` (from the generic test query) or `{{ test_metadata.kwargs.model }}` (from the test node), since that's really the thing we want.

Reply:
Great point on the 'where' @jtcohen6, we totally overlooked that!  We've added that in, but not pushed the change yet, while we wait to hear from the language team on how we might approach getting the non-model resources appropriately.